lms/services/h_operations.py

```diff
@@ -0,0 +1,33 @@
+from lms.services import HAPIError
+from lms.values import HUser
+
+
+__all__ = ["HypothesisOperationsService"]
+
+
+class HypothesisOperationsService:
+    """
+    High-level h API service.
+
+    The ``hapi`` service provides a low-level client for the Hypothesis "h"
+    API. This service provides higher-level operations (eg. fetching a user)
+    on top of that client.
+    """
+
+    def __init__(self, _context, request):
+        self.hapi_svc = request.find_service(name="hapi")
+        self.request = request
+
+    def fetch_user(self, username):
+        authority = self.request.registry.settings["h_authority"]
+        userid = f"acct:{username}@{authority}"
```

Comment:
Is there some way to avoid duplicating this string formatting? It's now in three places on this branch. If we change `HUser` to make `display_name` optional then you can do:

```python
>>> HUser("seanh", "hypothes.is").userid
'acct:seanh@hypothes.is'
```

Here's the diff to enable this (I had to move `display_name` to the bottom because optional args must come after all required ones; I also swapped the order of username and authority so that they make sense when used as positional args):

```diff
diff --git a/lms/values.py b/lms/values.py
index 64572dd1..c12c1761 100644
--- a/lms/values.py
+++ b/lms/values.py
@@ -27,6 +27,13 @@ class HUser(NamedTuple):
     usernames.
     """
 
+    username: str
+    """
+    Generated username for h user.
+
+    This is derived from the LMS userid.
+    """
+
     authority: str
     """
     The authority which the user belongs to.
@@ -34,16 +41,9 @@ class HUser(NamedTuple):
     This should always match ``settings["h_authority"]``
     """
 
-    display_name: str
+    display_name: str = ""
     """The display name for the user, generated from the LMS user's display name."""
 
-    username: str
-    """
-    Generated username for h user.
-
-    This is derived from the LMS userid.
-    """
-
     @property
     def userid(self):
         return f"acct:{self.username}@{self.authority}"
```